engines/myst3/myst3.cpp

```diff
@@ -1487,6 +1491,28 @@ bool Myst3Engine::canLoadGameStateCurrently() {
 	return _interactive;
 }
 
+void Myst3Engine::tryAutoSaving() {
+	if (!canLoadGameStateCurrently()) {
+		return; // Can't save right now, try again on the next frame
+	}
+
+	_lastSaveTime = _system->getMillis();
+
+	if (!_state->isAutoSaveAllowed()) {
+		return; // Can't autosave ever, try again after the next autosave delay
+	}
+
+	Common::String saveName = "Autosave";
+	Common::String fileName = saveName + ".M3S";
```

Comment:
The XBox version of the game has incompatible saves when compared to the PC/Mac version. In ResidualVM they have different filename extensions to set them apart. This function can be used to construct the filename with the proper extension:
https://github.com/residualvm/residualvm/blob/9484bbdd21a8e3c08c8e20ffe00b70b15c475ff3/engines/myst3/state.h#L439